Feature/final touches before 0.8.0

kubaplas opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Refactor**
	- Renamed entities for clarity and consistency: `ModuleWrapper` to `SubModule`, and `Variable` to `Var`.
- **Tests**
	- Ensured test suite accuracy post renaming of `Variable` to `Var` and `ModuleWrapper` to `SubModule.
- **Documentation**
	- Updated README and code comments to reflect recent changes for improved user and developer understanding.
- **Style**
	- Enhanced code readability and maintainability through adjustments in naming conventions and style.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

kubaplas committed:
Renamed ModuleWrapper -> SubModule

kubaplas committed:
Renamed Variable -> Var

kubaplas committed:
Unify crate versions and dependencies.

kubaplas commented:
Still missing:
- Remove odra-test from odra-vm

kubaplas committed:
Removed odra-test dependency from odra-vm

kubaplas committed:
Update template to handle new way of deploying code

kubaplas commented:
@kpob ready for review

zie1ony commented on Cargo.toml:
```diff
@@ -4,6 +4,7 @@ members = [
     "odra-vm",
     "core",
     "odra-macros",
+    "odra-casper/casper-client",
```
Why twice `odra-casper/casper-client`?
BTW, this name is a bit weird. Why don't we rename this crate to `odra-casper-rpc-client` in `odra-casper/rpc-client` directory?

kubaplas replied:
Ok

zie1ony commented on core/src/variable.rs:
```diff
@@ -8,22 +8,22 @@ use crate::prelude::*;
 use crate::{OdraError, UnwrapOrRevert};
```
File name remains `variable.rs`?

kubaplas replied:
Renamed

zie1ony commented on examples/Cargo.toml:
```diff
@@ -4,7 +4,7 @@ version = "0.8.0"
 
 [dependencies]
-odra = { path = "../odra", default-features = false }
+odra = { path = "../odra", version = "0.8.0", default-features = false }
```
No need to do this in `examples`, this crate is not being published.

kubaplas replied:
ok

kubaplas committed:
Codecoverage unfix

zie1ony commented on examples/src/features/collecting_events.rs:
```diff
@@ -3,8 +3,8 @@ use casper_event_standard::Event;
 use odra::{
-    module::{Module, ModuleWrapper},
-    Variable
+    module::{Module, SubModule},
```
Two things are wrong here.
1) `odra::module::Module` should not be imported. Code generated by macros use it, but it uses full path, so no need to import it. Please remove it from examples, and modules.
2) `SubModule` should be imported same way `Var` and `Mapping` so suggest it is `use odra::SubModule`.

kubaplas replied:
@zie1ony Doesn't work without importing Module:
```
error[E0599]: no method named `env` found for reference `&collecting_events::Machine` in the current scope
  --> src/features/collecting_events.rs:49:14
   |
49 |         self.env().emit_event(Info {
   |              ^^^ method not found in `&Machine`
   |
   = help: items from traits can only be used if the trait is in scope
help: some of the expressions' fields have a method of the same name
   |
49 |         self.e1.name.env().emit_event(Info {
   |              ++++++++
49 |         self.e2.name.env().emit_event(Info {
   |              ++++++++
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
   |
2  + use odra::module::Module;
   |
```

kubaplas replied:
I changed both:
```
use odra::{Submodule, Module}
```

zie1ony commented on examples/src/features/module_nesting.rs:
```diff
@@ -2,20 +2,20 @@ use casper_event_standard::Event;
 use odra::casper_event_standard;
```
I think this line might be avoided, if `casper_event_standard` is exported via `prelude`.

zie1ony commented on odra-vm/src/vm/odra_vm.rs:
```diff
@@ -616,7 +618,9 @@ mod tests {
 
     fn test_caller(entry_point_name: &str) -> EntryPointsCaller {
-        let env = odra_test::env();
+        let vm = OdraVm::new();
+        let host_env = OdraVmHost::new(vm);
+        let env = HostEnv::new(host_env);
```
I think it solves the problem :)

zie1ony commented on templates/module.rs.template:
```diff
@@ -1,13 +1,13 @@
+use odra::Var;
 
-/// A module definition. Each module struct consists Variables and Mappings
+/// A module definition. Each module struct consists Vars and Mappings
```
```suggestion
/// A module definition. Each module struct consists Vars and Mappings.
```

kubaplas replied:
There is a second part of this sentence line below

zie1ony commented on templates/module.rs.template:
```diff
@@ -36,12 +36,13 @@ impl #module_name {
     fn it_works() {
         let env = odra_test::env();
-        let mut contract = #module_nameDeployer::init(&env);
+        let mut contract = #module_nameHostRef::deploy(&env, NoInit);
```
```suggestion
        let mut contract = #module_nameHostRef::deploy(&env, NoArgs);
```
@kpob, first victim of `NoArgs` vs `NoInit`.

kpob replied:
It's recommended to read the documentation first. :) 
My idea was `EmptyArgs` or `DefaultArgs`. Now both start with `No` that might be a source of confusion.

kubaplas replied:
Fixed

kubaplas committed:
PR fixes.
Exposed Module and Submodule in `odra::`.
Renamed odra-casper-client to odra-casper-rpc-client.

kubaplas commented on core/src/lib.rs:
```diff
@@ -27,7 +27,7 @@ mod sequence;
 mod unwrap_or_revert;
 pub mod utils;
-mod variable;
+mod var;
```
No

kubaplas committed:
Hide Module, Event and casper_event_standard in prelude::*

kpob committed:
Remove `NoInit` (#348)

* Remove `NoInit`
